wm/client.go

```diff
@@ -0,0 +1,258 @@
+// +build linux,!ci
+
+package wm
+
+import (
+	"github.com/BurntSushi/xgb/xproto"
+	"github.com/BurntSushi/xgbutil/ewmh"
+	"github.com/BurntSushi/xgbutil/icccm"
+	"github.com/BurntSushi/xgbutil/xevent"
+	"github.com/BurntSushi/xgbutil/xprop"
+
+	"fyne.io/desktop"
+	"fyne.io/fyne"
+)
+
+type client struct {
+	id, win xproto.Window
+
+	framed    bool
+	iconic    bool
+	maximized bool
+	title     string
+	class     []string
+	command   string
+	iconName  string
+
+	allowedActions []string
+	stateHints     []string
+
+	fr *frame
+	wm *x11WM
+}
+
+func (s *stack) clientForWin(id xproto.Window) desktop.Window {
+	for _, w := range s.clients {
+		if w.(*client).id == id || w.(*client).win == id {
+			return w
+		}
+	}
+
+	return nil
+}
+
+func (cli *client) Decorated() bool {
+	return cli.framed
+}
+
+func (cli *client) Title() string {
+	if cli.title == "" {
```

Comment:
Why are these being cached? What happens if the hint changes?
It seems like a premature optimisation...